lowranceusr.cc

```diff
@@ -423,28 +960,397 @@ lowranceusr_parse_waypt(Waypoint* wpt_tmp)
 
   /* Waypoint Type (USER, TEMPORARY, POINT_OF_INTEREST) */
   short waypt_type = gbfgetint16(file_in);
-  if (global_opts.debug_level >= 1) {
-    printf(MYNAME " parse_waypt: waypt_type = %d\n",waypt_type);
+  if (global_opts.debug_level > 2) {
+    if (global_opts.debug_level == 99) {
+      printf(" %04x (%d)", (int)waypt_type, (int)waypt_type);
+    } else {
+      printf(MYNAME " parse_waypt: waypt_type = %d\n",waypt_type);
+    }
   }
 
   // Version 3 has a depth field here.
-  if (reading_version >= 3) {
+  if (reading_version == 3) {
     float depth_feet = gbfgetflt(file_in);
     if (std::abs(depth_feet - 99999.0)  > .1) {
```

Comment:
L973 implies the field is stored as a float.
L974 implies that the magic value +99999.0 indicates no data.
compare this with the comment at L1808-1809.